File: core/src/org/systempro/project/ui/Widget.java
package org.systempro.project.ui;

import com.badlogic.gdx.math.Vector2;

public abstract class Widget {
    Vector2 position;
    Size size;
    public Widget(){
        position=new Vector2(0,0);
        size=new Size(0,0);
    }

    public abstract void draw(WidgetRenderer renderer);
    public abstract Size calculateSize(Size maxSize);
    public abstract Vector2 calculateLocation(Vector2 parentLocation);
    public abstract void animate(float delta);
}

File: core/src/org/systempro/project/ui/Widget.scala
//package org.systempro.project.ui
//
//import com.badlogic.gdx.math.Vector2
//
//import scala.beans.BeanProperty
//
//abstract class Widget() {
//  var position:Vector2=null;
//  var size:Size=null;
//
//  def this(position:Vector2,size:Size)={
//    this();
//    this.position=position;
//    this.size=size;
//  }
//
//  def draw(renderer: WidgetRenderer): Unit
//
//  def calculateSize(maxSize: Size): Size
//
//  def calculateLocation(parentLocation: Vector2): Vector2
//
//  def animate(delta: Float): Unit
//}
